### PR TITLE
Palindrome_-Checke

### PR DESCRIPTION
**Pull request by Marwa-Sadat**
*Palindrome Checker*: 🪞✨ A simple tool that checks if a word, phrase, or sequence reads the same backward as forward. 🔁📝 It removes spaces and punctuation, ensuring an accurate check. 🔍✅ Perfect for linguistics enthusiasts, puzzle lovers, and anyone curious about symmetrical sequences! 🧩❤ Enhance your understanding of palindromes with ease and efficiency. 📚💡
Screenshot 2024-06-26 195426

### CODE REVIEW

**Review by zainabbehzad — CHANGES_REQUESTED** [review]
Hi @Marwa-Sadat
Excellent work You have done a great job implementing the project requirements.

✔Great design implemented.
✔Created a professional Repo.
✔Correct use of Gitflow.
✔Commit messages are meaningful.
✔Setup the linters correctly.
✔No linters errors.
✔You have followed and completed the Palindrome Checker and all requirements.
✔Your codes passes all FreeCodeCamp requiremenrs.
✔Created the pull request (From your  Palindrome-checker branch to the development branch you created a pull request and submitted the pull request link for review).
✔Good PR description.
✔Your repo has a professional README.md.
✔Your project design is responsive.
🧩Your code is clean.
  On index.html
* Your body spacing is not aligned correctly:
* All child elements should be two spaces ahead of their parent elements.

**Comment by ahzamir** [comment]
@Marwa-Sadat 

- You have to implement your partner's feedback and get their approval, then ask for a review from me.

**Review by zainabbehzad — APPROVED** [review]
Hi @Marwa-Sadat
Excellent work You have done a great job implementing the project requirements.

✔Great design implemented.
✔Created a professional Repo.
✔Correct use of Gitflow.
✔Commit messages are meaningful.
✔Setup the linters correctly.
✔No linters errors.
✔You have followed and completed the Palindrome Checker and all requirements.
✔Your codes passes all FreeCodeCamp requiremenrs.
✔Created the pull request (From your  Palindrome-checker branch to the development branch you created a pull request and submitted the pull request link for review).
✔Good PR description.
✔Your repo has a professional README.md.
✔Your project design is responsive.
✔Your code is clean.
  
  Status Approved✔

**Comment by Marwa-Sadat** [comment]
@ahzamir Hi sir hope you are well, please check my project .

File: README.md [hunk 1]
```diff
@@ -1,2 +1,117 @@
-# -Palindrome_-Checker
-*Palindrome Checker*: 🪞✨ A simple tool that checks if a word, phrase, or sequence reads the same backward as forward. 🔁📝
+
+
+markdown
```
Comment (ahzamir) [inline README.md]: Remove these 3 lines

File: README.md [hunk 2]
```diff
@@ -1,2 +1,117 @@
...
+3. [Contributing](#contributing)
+4. [License](#license)
+
+## Installation
+
+Follow the steps below to install and run the Palindrome Checker on your local machine.
+
+### Prerequisites
+
+- [Your Programming Language] installed on your machine.
```
Comment (ahzamir) [inline README.md:26]: - This is fine but it is like a template add your programming language here for example `[NodeJS] Installed on your machine.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,117 @@
...
+
+### Steps
+
+1. **Clone the Repository**:
+    bash
+    git clone https://github.com/yourusername/palindrome-checker.git
```
Comment (ahzamir) [inline README.md]: - Here add your project SSH link for now it is like an example

File: README.md [hunk 4]
```diff
@@ -1,2 +1,117 @@
...
+    cd palindrome-checker
+    
+
+2. **Install Dependencies**:
+    If there are any dependencies, list them here and provide installation instructions. For example:
+    bash
+    pip install -r requirements.txt
+    
+    Or, if there are no dependencies, you can skip this step.
```
Comment (ahzamir) [inline README.md]: Remove these template examples as you don't have dependencies on your project.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,117 @@
...
+
+3. **Run the Program**:
+    bash
+    python palindrome_checker.py
```
Comment (ahzamir) [inline README.md]: - Remove these 2 lines instead add `Open Index.html from root folder`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,117 @@
...
+    
+
+## Usage
+
+After installation, you can use the Palindrome Checker as follows:
+
+1. **Run the Program**:
+    bash
+    python palindrome_checker.py
+
```
Comment (ahzamir) [inline README.md]: - Remove this as you don't use python

File: README.md [hunk 7]
```diff
@@ -1,2 +1,117 @@
...
+## Example
+
+Here is an example of how to use the Palindrome Checker:
+
+bash
+$ python palindrome_checker.py
+Enter the string to check: A man a plan a canal Panama
+The string is a palindrome!
+
+
```
Comment (ahzamir) [inline README.md]: - Remove this example as this is for python and you used JS

File: README.md [hunk 8]
```diff
@@ -1,2 +1,117 @@
...
+## Demo
+
+![Demo](https://private-user-images.githubusercontent.com/168111110/344379010-90dc49e6-f19b-4c93-8fab-c1822c441425.png?)
```
Comment (ahzamir) [inline README.md]: - Add your own demo (Screenshot or Live deployed link)

File: README.md [hunk 9]
```diff
@@ -1,2 +1,117 @@
...
+## Contributing
+
+We welcome contributions to this project! If you have an idea for a new feature or have found a bug, please open an issue or submit a pull request. Follow the steps below to contribute:
+
+1. **Fork the Repository**:
+    Click the 'Fork' button at the top right corner of this repository.
+
+2. **Clone Your Fork**:
+    bash
+    git clone https://github.com/yourusername/palindrome-checker.git
```
Comment (ahzamir) [inline README.md]: - Add you SSH link here

File: README.md [hunk 10]
```diff
@@ -1,2 +1,117 @@
...
+    cd palindrome-checker
+    
+
+3. **Create a New Branch**:
+    bash
+    git checkout -b feature/your-feature-name
```
Comment (ahzamir) [inline README.md]: - Update this with your own feature branch name

**Review by ahzamir — CHANGES_REQUESTED** [review]
@Marwa-Sadat 

- Well done marwa, you have done a great job implementing lots of requirements and creating such a well-designed project but your README.md file needs some changes inorder to get the final approval check the comments below for more details:

image

- And don't forget to deploy your project

**Comment by Marwa-Sadat** [comment]
@ahzamir Hi sir I hope you are well, I have done all the feedback, please check my project.

**Review by ahzamir — APPROVED** [review]
### Hi @Marwa-Sadat 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text on it with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Roman Numeral Converter project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️Your repo have been reviewed by your partner
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @ahzamir in your question so I can receive the notification.**

**Comment by Marwa-Sadat** [comment]
@ahzamir thank you so much sir😊🙂

**Comment by ahzamir** [comment]
> @ahzamir thank you so much sir😊🙂

- You're welcome

**Pull request merged**